tests/core/test_db.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+import pytest_asyncio
+
+from lnbits.db import QueryValues
+
+
+@pytest_asyncio.fixture(autouse=True)
+async def json_test_table(db):
+    try:
+        await db.execute("CREATE TABLE test (data JSON)")
+        yield db
+    finally:
+        await db.execute("DROP TABLE test")
+
+
+@pytest.mark.asyncio
+async def test_sql_json(db):
+    obj = {"a": 3, "b": "bar", "c": {"nested": "d"}}
+    another_obj = {"a": 2, "b": "baz"}
+
+    await db.execute("INSERT INTO test VALUES(?)", (obj,))
+    await db.execute("INSERT INTO test VALUES(?)", (another_obj,))
+
+    values = QueryValues()
+    rows = await db.fetchall(
+        f"""
+        SELECT * FROM test WHERE {values.json_path('data', 'a', type_=int)} >= {values(obj["a"])}
+        """,
+        values,
+    )
+    assert len(rows) == 1
+    assert rows[0].data == obj
+
+    values = QueryValues()
+    rows = await db.fetchall(
+        f"SELECT * FROM test WHERE {values.json_path('data', 'c', 'nested')} = {values(obj['c']['nested'])}",
+        values,
+    )
+    assert len(rows) == 1
+    assert rows[0].data == obj
+
+    values = QueryValues()
+    rows = await db.fetchall(
+        f"SELECT * FROM test WHERE {values.json_path('data', 'b')} != {values(obj['b'])}",
+        values,
+    )
+    assert len(rows) == 1
+    assert rows[0].data == another_obj
+
+    update = {"b": "baz"}
```

Comment:
- please also add test with nested level update

Reply:
sqlite and postgres behave slightly different in that regard. sqlite does a deep merge while postgres does a shallow one

Reply:
- what does this mean, can you please provide some examples?
- it is important that the persisted data is identical, people will start by using SQLite then move to PostgreSQL. There should be no difference in behavior when switching.

Reply:
nested updates are inconsistent, so updating `{"a": {"b": "c"}}` with `{"a": {"d": "e"}}` results in `{"a": {"b": "c", "d": "e"}}` for sqlite but `{"a": {"d": "e"}}` for postgres.

Reply:
there is no simple way to achieve the same in postgres, so the update function should rather look like `json_set(col, attr, value)`